### ISSUE TITLE
require-return, require-param: don't warn if @type is present

### ISSUE THREAD

**Issue by golopot** [gajus/eslint-plugin-jsdoc#268]
If function is annotated by `@type` tag, then it does not need `@param` and `@returns`. Annotating a function by `@types` tag is handy when you need to write a long function type many times, for example, an express request handler. 

**code**: 
```js
/**
 * @typedef {(req: Request, res: Response, next: NextFunction) => void} RequestHandler
 */

/** @type {RequestHandler} */
function foo(req, res, next) {

}
```
**current behavior**:
Warns about missing params tag and missing returns tag 
**expected behavior**:
No warnings.
